### PR TITLE
kernel: refactor and cleanup parts of stringobj.{c,h}

### PR DESCRIPTION
**Pull request by fingolfin**

### CODE REVIEW

File: src/stringobj.h [hunk 1]
```diff
@@ -336,15 +338,16 @@ void MakeImmutableString(Obj str);
 
 static inline Obj MakeString(const Char * cstr)
```
Comment (stevelinton) [inline src/stringobj.h]: What's the rationale for this change? The original was shorter and not obviously less efficient.

Comment (fingolfin) [reply src/stringobj.h]: Your comment is on line 348/350, which actually was not changed by this PR; so I am not quite sure which change you are talking about?

Comment (ChrisJefferson) [reply src/stringobj.h]: I would guess the C_NEW_STRING below? Where I agree with the change because I'd like to eliminate C_NEW_STRING, as it's a bit too "magic" (it couldn't be implemented as a C function as it effectively invisibly takes a pointer to it's first argument)

Comment (fingolfin) [reply src/stringobj.h]: Ahhh, so the code below! My bad, I should have realized that.

Indeed, it is as @ChrisJefferson said: I am trying to get rid of `C_NEW_STRING`
